packages/rspack-dev-server/package.json

```
@@ -42,19 +43,38 @@
     "@types/connect-history-api-fallback": "1.5.4",
     "@types/express": "4.17.21",
     "@types/mime-types": "2.1.4",
+    "@types/express-serve-static-core": "4.19.5",
+    "@types/serve-static": "1.15.7",
     "@types/ws": "8.5.10",
     "jest-serializer-path": "^0.1.15",
-    "typescript": "5.0.2"
+    "prebundle": "^1.1.0",
+    "typescript": "5.0.2",
+    "webpack-dev-server": "5.0.4",
+    "tsc-alias": "^1.8.8"
   },
   "dependencies": {
     "chokidar": "^3.6.0",
     "connect-history-api-fallback": "^2.0.0",
     "express": "^4.19.2",
     "http-proxy-middleware": "^2.0.6",
     "mime-types": "^2.1.35",
-    "webpack-dev-middleware": "^7.4.2",
-    "webpack-dev-server": "5.0.4",
-    "ws": "^8.16.0"
+    "ws": "^8.16.0",
+    "webpack-dev-middleware": "7.4.2",
+    "ansi-html-community": "^0.0.8",
+    "bonjour-service": "^1.2.1",
+    "colorette": "2.0.19",
+    "compression": "^1.7.4",
+    "graceful-fs": "4.2.10",
+    "html-entities": "^2.4.0",
+    "ipaddr.js": "^2.1.0",
+    "launch-editor": "^2.6.1",
+    "open": "^10.0.3",
+    "p-retry": "^6.2.0",
+    "schema-utils": "^4.2.0",
+    "selfsigned": "^2.4.1",
+    "serve-index": "^1.9.1",
+    "sockjs": "^0.3.24",
+    "spdy": "^4.0.2"
   },
```

Have you considered leveraging [resolutions](https://pnpm.io/package_json#resolutions) to pin the indirect dependency `webpack-dev-middleware`? It would look look this:
```suggestion
  },
  "resolutions": {
    "webpack-dev-server/webpack-dev-middleware": "7.4.2"
  },
```

---

The users of resolution are consumers, not pkg publishers. 
We don't want all consumers to need resolution.